src/types.ts

```diff
@@ -209,7 +213,19 @@ export interface SessionData {
 }
 
 export interface Session {
-  data: SessionData,
+  data: SessionData;
   id: string;
   flowId: string;
 }
+
+export interface DetailedSession extends Session {
```

Comment:
It would be nice to eventually land on a naming convention for data transfer objects (i.e. objects with database fields) vs. domain objects (i.e. the data model) vs utility objects (i.e. unusually shaped objects from custom queries). "Detailed" as a prefix for data transfer objects feels fine for now IMO but suggestions welcome.

Reply:
I also personally prefer meaningful types with names, sometimes using the utility types `Pick<T>` and `Omit<T>` rather than just adding optional properties to types which can happen a lot for convenience. Going for a new named type here is the right move I think.